tests/std/tests/P0784R7_library_machinery/test.cpp

```diff
@@ -0,0 +1,147 @@
+        int_wrapper_copy output[4] = {5, 6, 7, 8};
+
+        const auto result = _Copy_backward_unchecked(begin(input), end(input), end(output));
+        static_assert(is_same_v<remove_const_t<decltype(result)>, int_wrapper_copy*>);
+        assert(result == begin(output));
+        assert(equal(begin(expected_copy), end(expected_copy), begin(output), end(output)));
+    }
+
+#if _HAS_CXX20 && defined(__cpp_constexpr_dynamic_alloc) && defined(__clang__)
+    { // _Uninitialized_copy_unchecked
+        int_wrapper_copy input[] = {1, 2, 3, 4};
+        int_wrapper_copy output[4];
+
+        const auto result = _Uninitialized_copy_unchecked(begin(input), end(input), begin(output));
+        static_assert(is_same_v<remove_const_t<decltype(result)>, int_wrapper_copy*>);
+        assert(result == end(output));
+        assert(equal(begin(expected_copy), end(expected_copy), begin(output), end(output)));
+    }
+#endif // _HAS_CXX20 && defined(__cpp_constexpr_dynamic_alloc)
+
+    { // _Move_unchecked
+        int_wrapper_move input[]   = {1, 2, 3, 4};
+        int_wrapper_move output[4] = {5, 6, 7, 8};
+
+        const auto result = _Move_unchecked(begin(input), end(input), begin(output));
+        static_assert(is_same_v<remove_const_t<decltype(result)>, int_wrapper_move*>);
+        assert(result == end(output));
+        assert(equal(begin(expected_move), end(expected_move), begin(output), end(output)));
+        if (is_constant_evaluated()) {
+            assert(equal(begin(input), end(input), begin(expected_after_move), end(expected_after_move)));
+        }
+    }
+
+    { // _Move_backward_unchecked
+        int_wrapper_move input[]   = {1, 2, 3, 4};
+        int_wrapper_move output[4] = {5, 6, 7, 8};
+
+        const auto result = _Move_backward_unchecked(begin(input), end(input), end(output));
+        static_assert(is_same_v<remove_const_t<decltype(result)>, int_wrapper_move*>);
+        assert(result == begin(output));
+        assert(equal(begin(expected_move), end(expected_move), begin(output), end(output)));
+        if (is_constant_evaluated()) {
+            assert(equal(begin(input), end(input), begin(expected_after_move), end(expected_after_move)));
+        }
+    }
+
+#ifdef __cpp_lib_concepts
+    { // _Move_backward_common
+        int_wrapper_move input[]   = {1, 2, 3, 4};
+        int_wrapper_move output[4] = {5, 6, 7, 8};
+
+        const auto result = ranges::_Move_backward_common(begin(input), end(input), end(output));
+        static_assert(is_same_v<remove_const_t<decltype(result)>, int_wrapper_move*>);
+        assert(result == begin(output));
+        assert(equal(begin(expected_move), end(expected_move), begin(output), end(output)));
+        if (is_constant_evaluated()) {
+            assert(equal(begin(input), end(input), begin(expected_after_move), end(expected_after_move)));
+        }
+    }
+#endif // __cpp_lib_concepts
+
+#if _HAS_CXX20 && defined(__cpp_constexpr_dynamic_alloc) && defined(__clang__)

```

Comment:
@miscco Removing the `&& defined(__clang__)` results in several test failures for `cl` -- are these because of a bug?  Or is there some reason these newly-added functions should not work for MSVC?  Apologies if I've just missed the conversation here.

Reply:
I believe this predates the turn to `_CONSTEXPR20_DYNALLOC`

Reply:
Turns out the errors I was seeing were due to a compiler bug -- reported here (DevCom-1333853) and Tracking Issue updated (#1566).